### ISSUE TITLE
cli(baseline): classify tests never set lower_is_worse, masking mi.* regression suppression

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#827]
## Summary

`baseline_tests.rs` never exercises a `lower_is_worse: true` violation through
`Baseline::classify`, so the suite does not catch that `classify` ignores
metric direction — for the `mi.*` (Maintainability Index) family a genuine
regression is silently classified `Covered` and dropped from the gate.

## Location

- `big-code-analysis-cli/src/baseline_tests.rs:11-40` (`v` / `v_hashed` helpers hardcode `lower_is_worse: false`)
- `big-code-analysis-cli/src/baseline_tests.rs:339-624` (every `classify_*` test)
- Production defect this gap masks: `big-code-analysis-cli/src/baseline.rs:557` (`Some(recorded) if v.value <= recorded => Coverage::Covered { recorded }`)

## Evidence

The two `Violation` constructors used by every classify test hardcode the
direction flag:

```rust
fn v(path: &str, function: &str, start_line: usize, metric: &'static str, value: f64) -> Violation {
    Violation {
        ...
        lower_is_worse: false,
        ...
    }
}
```

A workspace search confirms **no** test in `baseline_tests.rs` ever sets
`lower_is_worse: true` (the only `true` occurrences are in
`thresholds_tests.rs`). Meanwhile `classify` decides Covered-vs-Regressed
purely on `v.value <= recorded`, with no consultation of `v.lower_is_worse`:

```rust
Some(recorded) if v.value.is_nan() => Coverage::Regressed { recorded },
Some(recorded) if v.value <= recorded => Coverage::Covered { recorded },
Some(recorded) => Coverage::Regressed { recorded },
```

For a lower-is-worse metric (`mi.original`, `mi.sei`, `mi.visual_studio`),
a value that *drops* below the recorded baseline is a genuine regression, but
`v.value <= recorded` is true, so `classify` returns `Covered`. At the gate
(`commands.rs:778-783`) `Coverage::Covered` violations are filtered out of the
kept set, so the regression is silently dropped.

This is the same direction-ignoring defect already filed for the diff layer in
#825 (`baseline_diff.rs::compute`), but here it lives in the baseline gate
itself. #825 explicitly notes its own test gap ("uses direction-agnostic metric
names, so the inversion for `mi.*` is unguarded") — the identical blind spot
exists here, one layer deeper and with a worse consequence (a hidden
regression rather than a mislabelled bucket).

## Expected Behavior

The test suite should cover `classify` with a `lower_is_worse: true` violation
in both directions:

- value below recorded (worsened for lower-is-worse) -> `Regressed`
- value above recorded (improved for lower-is-worse) -> `Covered`

so the gate's direction handling is pinned and a future fix to `classify` is
guarded.

## Actual Behavior

No classify test sets `lower_is_worse: true`. The current `classify`
implementation would fail a correctness-direction test (a worsened `mi.*`
returns `Covered`), so the missing coverage is actively masking a real
gate-suppression bug.

## Impact

`bca check` against a baseline containing `mi.*` entries silently drops
real Maintainability-Index regressions instead of surfacing them as
`Regressed`. Because the test suite never constructs a lower-is-worse
violation, the defect is invisible to CI. This is a bug-masking test gap,
not merely a missing nicety.

## Resolution (commit `4de5f41d`)

`Baseline::classify` now branches the Covered ratchet on
`v.lower_is_worse`: for the lower-is-worse `mi.*` family the covered
condition inverts to `value >= recorded`, so a value dropping below the
recorded baseline classifies `Regressed` instead of being silently
dropped as `Covered`. The NaN guard still classifies `Regressed`, and the
higher-is-worse path (`value <= recorded`) is unchanged.

Coverage: added a `v_low` lower-is-worse `Violation` constructor plus
`classify_lower_is_worse_drop_is_regressed` and
`classify_lower_is_worse_rise_is_covered`, both verified to fail against
the pre-fix `v.value <= recorded` line via test-via-revert. Existing
higher-is-worse classify tests are unchanged. Left open for your review.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

`Baseline::classify` ratchets on `v.value <= recorded` unconditionally, never
consulting `v.lower_is_worse`. The `baseline_tests.rs` `v`/`v_hashed` helpers
hardcode `lower_is_worse: false`, so no classify test can observe the
direction-inverted behavior. The result is two coupled defects: a production
gate that drops `mi.*` regressions, and a test suite that cannot see it.

### Implementation Steps

1. Production fix (separate from the test gap, but the tests below pin it):
   in `classify`, branch the ratchet on `v.lower_is_worse` — for a
   lower-is-worse metric the Covered condition is `v.value >= recorded`, and
   the NaN guard still classifies `Regressed`. Mirror the inversion already
   present in `thresholds.rs:818`.
2. Add a `lower_is_worse`-aware variant of the `v` helper (or a parameter)
   so classify tests can construct both directions without duplicating the
   full struct literal.

### Tests to Add/Update

- `classify_lower_is_worse_drop_is_regressed`: a `lower_is_worse: true`
  violation whose value is *below* the recorded baseline classifies
  `Regressed { recorded }` (currently returns `Covered` — verify the test
  fails against today's `classify`).
- `classify_lower_is_worse_rise_is_covered`: same metric, value *above*
  recorded classifies `Covered`.
- Keep the existing higher-is-worse tests unchanged to confirm the default
  direction is preserved.
- Verify-via-revert: confirm the new tests fail against the pre-fix
  `v.value <= recorded` line, per .claude/rules/testing.md.

### Verification

- [ ] `cargo test -p big-code-analysis-cli` passes
- [ ] `cargo clippy -p big-code-analysis-cli` clean
- [ ] Manual verification: write a baseline recording an `mi.original`
      value, then run `bca check` against source whose MI dropped below it;
      confirm the regression is now surfaced (`[regr ...]`) rather than
      silently covered.
- [ ] Cross-reference #825 (same direction defect in `baseline_diff.rs`) so
      both layers are fixed consistently.

**Comment by dekobon** [comment]
## Resolution (commit `4de5f41d`)

`Baseline::classify` now branches the Covered ratchet on
`v.lower_is_worse`: for the lower-is-worse `mi.*` family the covered
condition inverts to `value >= recorded`, so a value dropping below the
recorded baseline classifies `Regressed` instead of being silently
dropped as `Covered`. The NaN guard still classifies `Regressed`, and the
higher-is-worse path (`value <= recorded`) is unchanged.

Coverage: added a `v_low` lower-is-worse `Violation` constructor plus
`classify_lower_is_worse_drop_is_regressed` and
`classify_lower_is_worse_rise_is_covered`, both verified to fail against
the pre-fix `v.value <= recorded` line via test-via-revert. Existing
higher-is-worse classify tests are unchanged. Left open for your review.
